owl/SOMA-IO.owl

```diff
@@ -83,38 +554,454 @@
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#PluginSpecification">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#API_Specification"/>
+        <rdfs:subClassOf>
+            <owl:Class>
+                <owl:intersectionOf rdf:parseType="Collection">
+                    <owl:Restriction>
+                        <owl:onProperty rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#definesRole"/>
+                        <owl:someValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#HostRole"/>
+                    </owl:Restriction>
+                    <owl:Restriction>
+                        <owl:onProperty rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#definesRole"/>
+                        <owl:someValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#PluginRole"/>
+                    </owl:Restriction>
+                </owl:intersectionOf>
+            </owl:Class>
+        </rdfs:subClassOf>
+        <rdfs:label xml:lang="en">Plugin Specification</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#Programming_Language -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Programming_Language">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Computer_Language"/>
         <rdfs:subClassOf>
             <owl:Restriction>
-                <owl:onProperty rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#classifies"/>
-                <owl:allValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#SoftwareImplementation"/>
+                <owl:onProperty rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#givesMeaningTo"/>
+                <owl:allValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Computer_Code"/>
             </owl:Restriction>
         </rdfs:subClassOf>
-        <rdfs:comment>A concept that classifies software objects.</rdfs:comment>
-        <rdfs:label>Software type</rdfs:label>
+        <rdfs:comment xml:lang="en">Any Programming Language, including both human-readable like Java and non-human-readable languages like binary machine code.</rdfs:comment>
+        <rdfs:label xml:lang="en">Programming Language</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#QueryEngine -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#QueryEngine">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#SoftwareRole"/>
+        <rdfs:comment xml:lang="en">A Query Engine is a Software that can answer some queries.</rdfs:comment>
+        <rdfs:label xml:lang="en">Query engine</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#Reasoner -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Reasoner">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#SoftwareRole"/>
```

Comment:
Is SoftwareRole, as opposed to Software, really intended here?

Reply:
yes, see above